quill-cassandra/src/main/scala/io/getquill/source/cassandra/CassandraSyncSource.scala

```diff
@@ -0,0 +1,42 @@
+package io.getquill.source.cassandra
+
+import io.getquill.source.sql.SqlSource
+import io.getquill.naming.NamingStrategy
+import com.datastax.driver.core.Row
+import com.datastax.driver.core.BoundStatement
+import com.datastax.driver.core.Session
+import scala.collection.JavaConversions._
+import com.datastax.driver.core.ResultSet
+import com.datastax.driver.core.BatchStatement
+import com.datastax.driver.core.Cluster
+import scala.util.Try
+
+class CassandraSyncSource[N <: NamingStrategy]
+    extends SqlSource[CqlIdiom, N, Row, BoundStatement]
+    with Encoders
+    with Decoders {
+
+  protected val session: Session = ClusterSession(config)
+
+  def close = session.close
+
+  def probe(cql: String) =
+    Try(session.prepare(cql))
+
+  def execute(cql: String): ResultSet =
+    session.execute(session.prepare(cql).bind)
+
+  def execute(cql: String, bindList: List[BoundStatement => BoundStatement]): ResultSet = {
+    val ps = session.prepare(cql)
+    val batch = new BatchStatement
+    for (bind <- bindList)
+      batch.add(bind(ps.bind))
+    session.execute(batch)
+  }
+
+  def query[T](cql: String, bind: BoundStatement => BoundStatement, extractor: Row => T): List[T] = {
+    val stmt = session.prepare(cql).bind
```

Comment:
You should use a cache of PreparedStatement as they are prepared in the cluster; if you try to execute the same query again, the driver will complain that your are prepare the same query again and that it could have an impact on performance; for further information check this [link](https://github.com/datastax/java-driver/blob/3.0/features/statements/prepared/README.md#prepared-statements).

The same applies to every other method where you are preparing statements.